discretisedfield/field_new.py

```diff
@@ -77,8 +77,59 @@ def __init__(
     #    self._subregions = subregions
 
     @classmethod
-    def from_xarray(cls):  # -> still required (?)
-        raise NotImplementedError()
+    def from_xarray(cls, xa: xr.DataArray):
+
+        if not isinstance(xa, xr.DataArray):
+            raise TypeError("Argument must be a xr.DataArray.")
+
+        if len(xa.coords) != xa.data.ndim:
+            raise AttributeError(
+                "The xr.DataArray must have the same number of coordinates as"
+                " the axes of underlying data array."
+            )
+
+        for i in xa.coords.dims[:-1]:
+            if xa[i].data.size > 1 and not np.allclose(
+                np.diff(xa[i].data), np.diff(xa[i].data).mean()
+            ):
+                raise ValueError(f"Coordinates of {i} must be equally spaced.")
+
+        try:
+            cell = xa.attrs["cell"]
```

Comment:
Do we ensure here that cell makes sense? For instance, there could be some garbage in `cell` attribute passed via xarray.

Reply:
The value of `cell` is passed to `df.Mesh` below. Hence, if it is garbage, it will throw as error from the mesh class. Would you like to have an extra check here?

Reply:
Good point!